### PR TITLE
A few fixes for API daily test execution

### PR DESCRIPTION
**Pull request by lanej0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

A few odd fixes to improve the reliability of running API tests on daily

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Actions > Smoke Test Daily
2. Create a new run based off of this branch
3. Ensure that tests pass

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/api-core-tests/tests/shipping/shipping-zones.test.js [hunk 1]
```diff
@@ -171,8 +171,14 @@ test.describe.serial( 'Shipping zones API tests', () => {
...
 
-		await expect( putResponseStateOnlyJSON ).toHaveLength(0);
+		// running on external hosts, this can be 0 or 1
+		const responseLength = await putResponseStateOnlyJSON.length;
 
+		if ( responseLength == 0 ) {
+			await expect( putResponseStateOnlyJSON ).toHaveLength( 0 );
+		} else {
+			await expect( putResponseStateOnlyJSON ).toHaveLength( 1 );
+		}
```
Comment (alopezari) [inline plugins/woocommerce/tests/api-core-tests/tests/shipping/shipping-zones.test.js]: I think we can do the same in fewer lines with:
```
const responseLength = await putResponseStateOnlyJSON.length;
await expect(putResponseStateOnlyJSON).toHaveLength(responseLength === 0 ? 0 : 1);
```

Also, if we just want to check that `putResponseStateOnlyJSON` is 0 or 1, we might even simplify it with [expect(value).toContain()](https://playwright.dev/docs/api/class-genericassertions#generic-assertions-to-contain-2):
```
const responseLength = await putResponseStateOnlyJSON.length;
expect([0, 1]).toContain(responseLength);
```

Or even:
```
expect([0, 1]).toContain(putResponseStateOnlyJSON.length);
```

Note: I haven't tried this approach yet, we'd need to confirm that it works (I just suggested it based on the Playwright documentation).

**Review by alopezari — COMMENTED** [review]
Thanks for working on this @lanej0!

I just added a refactor suggestion. Apart from that, I tried executed the Smoke tests daily in this branch but the API tests failed 2 times in a row :(

This is the [log](https://github.com/woocommerce/woocommerce/actions/runs/6324046651/job/17174699375):
```
  1 failed
    [Chrome] › settings/settings-crud.test.js:661:3 › Settings API tests: CRUD › Batch Update a settings option › can batch update settings options 
  1 flaky
    [Chrome] › settings/settings-crud.test.js:276:3 › Settings API tests: CRUD › List all settings options › can retrieve all general settings 
```

At first glance, it doesn't look related to this PR but I wanted to check it with you just in case.

File: plugins/woocommerce/tests/api-core-tests/tests/shipping/shipping-zones.test.js (hunk 1 above)
Comment (lanej0) [reply plugins/woocommerce/tests/api-core-tests/tests/shipping/shipping-zones.test.js]: I like it!  That's way more succinct than the `if...else...`.

**Comment by lanej0** [comment]
The failures (were) unrelated, yes.  I was having some other test failures, but I merged in trunk, and got a [clean result](https://github.com/woocommerce/woocommerce/actions/runs/6425295284/job/17447627690) after that.

**Review by alopezari — APPROVED** [review]
Cool, thanks! I'll approve it 👍

**Pull request merged**